### PR TITLE
Palo Alto Prisma Cloud Data Connector

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add data connector
  - add workbook
  - add detection rules and hunting queries

### CODE REVIEW

**Review by v-ampami — CHANGES_REQUESTED** [review]

File: Solutions/PaloAltoPrismaCloud/Data Connectors/PrismaCloud_API_FunctionApp.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        },
+        {
+            "title": "",
+            "description": "**2. Configure the Function App**\n\n1. In the Function App, select the Function App Name and select **Configuration**.\n2. In the **Application settings** tab, select **+ New application setting**.\n3. Add each of the following application settings individually, with their respective string values (case-sensitive): \n\t\tPrismaCloudAPIUrl\n\t\tPrismaCloudAccessKeyID\n\t\tPrismaCloudSecretKey\n\t\tAzureSentinelWorkspaceId\n\t\tAzureSentinelSharedKey\n\t\tlogAnalyticsUri (Optional)\n - Use logAnalyticsUri to override the log analytics API endpoint for dedicated cloud. For example, for public cloud, leave the value empty; for Azure GovUS cloud environment, specify the value in the following format: https://WORKSPACE_ID.ods.opinsights.azure.us. \n4. Once all application settings have been entered, click **Save**."
```
Comment (v-ampami) [inline Solutions/PaloAltoPrismaCloud/Data Connectors/PrismaCloud_API_FunctionApp.json:141]: Sample loganalyticsuri should not be clickable. Can you please update https://WORKSPACE_ID.ods.opinsights.azure.us between Grave Accent symbol( ` ).

image

**Review by v-ampami — CHANGES_REQUESTED** [review]

**Review by shainw — COMMENTED** [review]
Approved from the queries perspective

File: Parsers/PaloAltoPrismaCloud/PaloAltoPrismaCloud.txt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        alertRules_s=column_ifexists('alertRules_s', ''),
+        riskDetail_riskScore_score_d=column_ifexists('riskDetail_riskScore_score_d', ''),
+        riskDetail_riskScore_maxScore_d=column_ifexists('riskDetail_riskScore_maxScore_d', ''),
+        riskDetail_rating_s=column_ifexists('riskDetail_rating_s', ''),
+        riskDetail_score_s=column_ifexists('riskDetail_score_s', ''),
+        status_s=column_ifexists('status_s', ''),
+        firstSeen_d=column_ifexists('firstSeen_d', ''),
+        lastSeen_d=column_ifexists('lastSeen_d', ''),
+        alertTime_d=column_ifexists('alertTime_d', ''),
+        resource_id=iff(isnotempty(column_ifexists('resource_id_s', '')), column_ifexists('resource_id_s', ''), column_ifexists('resource_id_g', ''))
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Parsers/PaloAltoPrismaCloud/PaloAltoPrismaCloud.txt]: resource_id can be formatted as string and as guid, but I provided only "string" records in the sample data. I will add "guid" records to sample data so we will have resource_id_g and resource_id_s in the table.

File: DataConnectors/PaloAltoPrismaCloud/PrismaCloud_API_FunctionApp.json [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        {
+            "metricName": "Prisma Cloud audit logs",
+            "legend": "PaloAltoPrismaCloudAudit_CL",
+            "baseQuery": "PaloAltoPrismaCloudAudit_CL"
+        }
+    ],
+    "sampleQueries": [
+        {
+            "description" : "All Prisma Cloud alerts",
+            "query": "PaloAltoPrismaCloudAlert\n| sort by TimeGenerated desc"
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply DataConnectors/PaloAltoPrismaCloud/PrismaCloud_API_FunctionApp.json]: updated

File: DataConnectors/PaloAltoPrismaCloud/PrismaCloud_API_FunctionApp.json [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        },
+        {
+            "description" : "All Prisma Cloud audit logs",
+            "query": "PaloAltoPrismaCloudAudit\n| sort by TimeGenerated desc"
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply DataConnectors/PaloAltoPrismaCloud/PrismaCloud_API_FunctionApp.json]: updated

File: Solutions/PaloAltoPrismaCloud/Data Connectors/PrismaCloud_API_FunctionApp.json (hunk 1 above)
Comment (vu-socprime) [reply Solutions/PaloAltoPrismaCloud/Data Connectors/PrismaCloud_API_FunctionApp.json]: updated

File: Parsers/PaloAltoPrismaCloud/PaloAltoPrismaCloud.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Parsers/PaloAltoPrismaCloud/PaloAltoPrismaCloud.txt]: updated

**Review by v-jayakal — APPROVED** [review]

File: Solutions/PaloAltoPrismaCloud/Parsers/PaloAltoPrismaCloud.txt [hunk 5]
```diff
@@ -0,0 +1,167 @@
+// Usage Instruction :
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name and alias as PaloAltoPrismaCloud.
+// Function usually takes 10-15 minutes to activate. You can then use function alias from any other queries (e.g. PaloAltoPrismaCloud | take 10).
+// Reference : Using functions in Azure monitor log queries : https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+let Audit_view = view () { 
+    PaloAltoPrismaCloudAudit_CL
+| where isnotempty(user_g)
```
Comment (v-maudan) [inline Solutions/PaloAltoPrismaCloud/Parsers/PaloAltoPrismaCloud.txt:7]: In sample data there is no Guid provided for user and resourceName, hence it is failing at user_g and resourceName_g  to save parser. I have updated some sample GUID in **PaloAltoPrismaCloudAudit_CL.json** of user and resourceName fields to solve this issue.

can you please check and let us know if there is any change.

Thanks,
Mallikarjun

Comment (vu-socprime) [reply Solutions/PaloAltoPrismaCloud/Parsers/PaloAltoPrismaCloud.txt:7]: Fields resourceName and user can be _s and _g. I will update sample data. Also the line in parser
`| where isnotempty(user_g)`
is excessive and must be removed.
Could I fix it in a new PR as this PR is already merged?

Comment (vu-socprime) [reply Solutions/PaloAltoPrismaCloud/Parsers/PaloAltoPrismaCloud.txt:7]: @v-maudan - sample data looks ok after your change. We only need to remove this line from parser:
`| where isnotempty(user_g)`

Comment (v-maudan) [reply Solutions/PaloAltoPrismaCloud/Parsers/PaloAltoPrismaCloud.txt:7]: @vu-socprime , Thanks for your confirmation, please raise new PR.

Comment (vu-socprime) [reply Solutions/PaloAltoPrismaCloud/Parsers/PaloAltoPrismaCloud.txt:7]: @v-maudan PR created - https://github.com/Azure/Azure-Sentinel/pull/2116

File: Solutions/PaloAltoPrismaCloud/Analytic Rules/PaloAltoPrismaCloudHighRiskScoreAlert.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+id: 617b02d8-0f47-4f3c-afed-1926a45e7b28
+name: Palo Alto Prisma Cloud - Maximum risk score alert
```
Comment (v-maudan) [inline Solutions/PaloAltoPrismaCloud/Analytic Rules/PaloAltoPrismaCloudHighRiskScoreAlert.yaml:2]: @vu-socprime ,"PaloAltoPrismaCloudMaxRiskScoreAlert.yaml" analytics rule having same name. Could you please update name of this Analytics rule to"Palo Alto Prisma Cloud - **High** risk score alert".   Thanks!

Comment (vu-socprime) [reply Solutions/PaloAltoPrismaCloud/Analytic Rules/PaloAltoPrismaCloudHighRiskScoreAlert.yaml:2]: Updated in PR https://github.com/Azure/Azure-Sentinel/pull/2116

File: Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudUpdatedResources.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    dataTypes:
+      - PaloAltoPrismaCloud
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1133
+query: |
+  PaloAltoPrismaCloud
+  | where TimeGenerated > ago(24h)
+  | Reason =~ 'RESOURCE_UPDATED'
```
Comment (v-maudan) [inline Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudUpdatedResources.yaml:17]: @vu-socprime , Please correct the query by adding  where before Reason to fix.
| **where** Reason =~ 'RESOURCE_UPDATED'

File: Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudTopResources.yaml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  - InitialAccess
+relevantTechniques:
+  - T1133
+query: |
+  PaloAltoPrismaCloud
+  | where TimeGenerated > ago(24h)
+  | where Reason =~ 'NEW_ALERT'
+  | summarize count() by ResourceName
+  | order by count_ desc
+  | extend AccountCustomEntity = UserName
```
Comment (v-maudan) [inline Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudTopResources.yaml:20]: @vu-socprime ,  This query is not working , In order to fix this please move" | extend AccountCustomEntity = UserName" before "| Summarize ....".[PFB updated query]

  PaloAltoPrismaCloud
  | where TimeGenerated > ago(24h)
  | where Reason =~ 'NEW_ALERT'
  **| extend AccountCustomEntity = UserName**
  | summarize count() by ResourceName
  | order by count_ desc

File: Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudUpdatedResources.yaml (hunk 7 above)
Comment (v-maudan) [inline Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudUpdatedResources.yaml:17]: @vu-socprime , This query is failing , To fix this please use where before Reason.[PFB updated query]

| **where** Reason =~ 'RESOURCE_UPDATED'

File: Solutions/PaloAltoPrismaCloud/Analytic Rules/PaloAltoPrismaCloudHighRiskScoreAlert.yaml (hunk 6 above)
Comment (v-maudan) [reply Solutions/PaloAltoPrismaCloud/Analytic Rules/PaloAltoPrismaCloudHighRiskScoreAlert.yaml:2]: @vu-socprime , Two hunting queries are failing, I have given comment for same, can you please fix those in same PR. Thanks!

File: Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudUpdatedResources.yaml (hunk 7 above)
Comment (vu-socprime) [reply Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudUpdatedResources.yaml:17]: fixed

File: Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudTopResources.yaml (hunk 8 above)
Comment (vu-socprime) [reply Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudTopResources.yaml:20]: updated in new PR - https://github.com/Azure/Azure-Sentinel/pull/2132

Comment (v-maudan) [reply Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudTopResources.yaml:20]: @vu-socprime , We have already merged this fix, can you please check and confirm . [PR 2126](https://github.com/Azure/Azure-Sentinel/pull/2126).

Comment (vu-socprime) [reply Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudTopResources.yaml:20]: We need to change UserName to ResourceName. I have made changes in https://github.com/Azure/Azure-Sentinel/pull/2132

Comment (v-maudan) [reply Solutions/PaloAltoPrismaCloud/Hunting Queries/PaloAltoPrismaCloudTopResources.yaml:20]: ok Thanks !

**Pull request merged**